libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -40,6 +40,8 @@ logScope:
 declareCounter(libp2p_gossipsub_failed_publish, "number of failed publish")
 declareCounter(libp2p_gossipsub_invalid_topic_subscription, "number of invalid topic subscriptions that happened")
 declareCounter(libp2p_gossipsub_duplicate_during_validation, "number of duplicates received during message validation")
+declareCounter(libp2p_gossipsub_idontwant_saved_messages, "number of duplicates avoided by idontwant")
+declareCounter(libp2p_gossipsub_saved_bytes, "bytes saved by gossipsub optimizations", labels=["kind"])
```

Comment:
or maybe not , hmm.. might be interesting indeed

Reply:
the structural difference between bytes and messages is ugly though, so maybe separate metrics instead of labels to keep the structure consistent

Reply:
The `libp2p_gossipsub_saved_bytes` is always in byte
You mean having a `libp2p_gossipsub_saved_messages` with two kinds instead of `libp2p_gossipsub_duplicate_during_validation` & `libp2p_gossipsub_idontwant_saved_messages`
?

Reply:
I mean having `libp2p_gossipsub_idontwant_saved_bytes` instead of a label

Reply:
What I find "cool" with the `libp2p_gossipsub_saved_bytes` is that you can add the total to a graph and you're "future proof"

Reply:
But won't argue over it if you prefer a different metric

Reply:
alright . hm . yeah no strong opinion either, let's leave it then